ffc/formatting.py

```diff
@@ -177,6 +178,11 @@ def _generate_includes(includes, parameters):
     s = set(default_h_includes) | includes
 
     # s2 = external_includes - s
+    includes_h = "\n".join(sorted(s)) + "\n" if s else ""
 
-    includes = "\n".join(sorted(s)) + "\n" if s else ""
-    return includes
+    # This should really be set by the backend
+    if "scalar_type" in parameters.keys():
+        if parameters["scalar_type"] == "double complex":
+            includes_c = "#include<complex.h>" + "\n"
+
```

Comment:
Maybe a bit more like:
```
scalar_type = parameters.get("scalar_type")
if scalar_type is "scalar_type": 
    includes_c = "#include<complex.h>\n"
```

Reply:
you mean `if scalar_type is "double complex":`

Reply:
Much better.
I'll keep `==` instead of `is`, otherwise the if would never be true.